ibis/backends/duckdb/__init__.py

```diff
@@ -340,47 +340,67 @@ def read_parquet(
         ir.Table
             The just-registered table
         """
-        import sqlalchemy_views as sav
-
         source_list = normalize_filenames(source_list)
 
-        if any(source.startswith("s3://") for source in source_list):
-            if len(source_list) > 1:
-                raise ValueError("only single s3 paths are supported")
+        table_name = table_name or f"ibis_read_parquet_{next(pa_n)}"
 
-            import pyarrow.dataset as ds
-
-            dataset = ds.dataset(source_list[0])
-            table_name = table_name or f"ibis_read_parquet_{next(pa_n)}"
-            self._load_extensions(["httpfs"])
-            # We don't create a view since DuckDB special cases Arrow Datasets
-            # so if we also create a view we end up with both a "lazy table"
-            # and a view with the same name
-            with self.begin() as con:
-                # DuckDB normally auto-detects Arrow Datasets that are defined
-                # in local variables but the `dataset` variable won't be local
-                # by the time we execute against this so we register it
-                # explicitly.
-                con.connection.register(table_name, dataset)
-        else:
-            if any(
-                source.startswith(("http://", "https://")) for source in source_list
+        # Default to using the native duckdb parquet reader
+        # If that fails because of auth issues, fall back to ingesting via
+        # pyarrow dataset
+        try:
+            self._read_parquet_duckdb_native(source_list, table_name, **kwargs)
+        except sa.exc.OperationalError as e:
+            if ("403 (Forbidden)" in repr(e)) or (
```

Comment:
Is there nothing more specific coming from duckdb that we might be able to catch?

If not, is there any reason to match the exception at all?

Reply:
I think it's useful to distinguish errors due to permission/url support issues (where falling back to the more flexible but slower option is needed) from errors due to:

- A file not existing
- A file being corrupted
- Some other IO failure (e.g. 500 errors or something)

Especially for network requests where there might be a bit of a lag before an error happens, falling back on any error seems risky. If the errors like these aren't `OperationalErrors` though then we might be fine to drop the message filtering.

Reply:
I see, this is fine for now, though I wonder if we should aggressively remove this once pyarrow 12 is out and just tell people they have to upgrade pyarrow if they want better performance.

Reply:
It's [valid to raise an exception subclass](https://peps.python.org/pep-0249/#exceptions), so we can ask them to isolate those errors upstream.

Reply:
That's not blocking this PR though.